app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,38 @@
+class LikesController < ApplicationController
+  include ActionView::RecordIdentifier
+  def update
+    if likes.exists?
+      likes.destroy_all
+    elsif current_user.likes.count < User::LIKE_LIMIT
+      likes.first_or_create(user: current_user)
+    else
+      errors = "You've reached the like limit"
```

Comment:
I'd try to later turn this into a model validation and then check for resource .valid and render all the errors or find an alternative way to render the validation errors